### PR TITLE
Feature/on demand scenarios

### LINKED ISSUE #1048: XATTRs: On demand processing of docs
Sync Gateway to process document mutations in real-time for requests (e.g. via SG Get()) of documents that have not yet been processed

### PR DESCRIPTION
**Pull request by sridevi-15**
#### Fixes #1048 

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- Add xattrs test with import disabled which does on demand processing
- First sceneario mentioned in the ticket is covered in new test
   Second scenario mentioned in the ticket is mostly covered in existing test and add few steps for have full coverage

### CODE REVIEW

File: testsuites/syncgateway/functional/tests/test_xattrs.py [hunk 1]
```diff
@@ -153,6 +155,142 @@ def test_olddoc_nil(params_from_base_test_setup, sg_conf_name):
...
+    bucket_name = 'data-bucket'
+    sg_db = 'db'
+    cbs_host = host_for_url(cbs_url)
+
+    # This test should only run when using xattr meta storage
+    if not xattrs_enabled:
+        pytest.skip('XATTR tests require --xattrs flag')
+
+    # This test should only run when mode is CC
+    if mode == "DI":
```
Comment (raghusarangapani) [inline testsuites/syncgateway/functional/tests/test_xattrs.py]: This probably needs to be lowercase? Test it out with lowercase in the pytest --mode parameter

File: testsuites/syncgateway/functional/tests/test_xattrs.py [hunk 2]
```diff
@@ -179,6 +317,9 @@ def test_on_demand_import_of_external_updates(params_from_base_test_setup, sg_co
     if not xattrs_enabled:
         pytest.skip('XATTR tests require --xattrs flag')
 
+    if mode == "DI":
```
Comment (raghusarangapani) [inline testsuites/syncgateway/functional/tests/test_xattrs.py]: lower case?

Comment (sridevi-15) [reply testsuites/syncgateway/functional/tests/test_xattrs.py]: modified to lower case

File: testsuites/syncgateway/functional/tests/test_xattrs.py (hunk 1 above)
Comment (sridevi-15) [reply testsuites/syncgateway/functional/tests/test_xattrs.py]: yeap, modified to lower case to be in consitent with other tests

File: resources/sync_gateway_configs/xattrs/no_import_cc.json [hunk 3]
```diff
@@ -9,6 +9,7 @@
     "log": ["*"],
     "databases":{
         "db":{
+            {{ xattrs }}
```
[the remark this answers is not in the document]
Comment (sridevi-15) [reply resources/sync_gateway_configs/xattrs/no_import_cc.json]: removed xattrs variable from sg config as it is not required

**Comment by raghusarangapani** [comment]
Looks ok. Once a tracking ticket for enabling ssl is logged, this can be merged

**Pull request merged**